pkg/git/libgit2/checkout.go

```diff
@@ -249,11 +340,26 @@ type CheckoutCommit struct {
 func (c *CheckoutCommit) Checkout(ctx context.Context, path, url string, opts *git.AuthOptions) (_ *git.Commit, err error) {
 	defer recoverPanic(&err)
 
+	remoteCallBacks := RemoteCallbacks(ctx, opts)
+
+	if managed.Enabled() {
+		if opts.TransportOptionsURL == "" {
+			return nil, fmt.Errorf("can't use managed transport without a valid transport auth id.")
+		}
+		managed.AddTransportOptions(opts.TransportOptionsURL, managed.TransportOptions{
+			TargetURL:    url,
+			AuthOpts:     opts,
+			ProxyOptions: &git2go.ProxyOptions{Type: git2go.ProxyTypeAuto},
+		})
+		url = opts.TransportOptionsURL
+		remoteCallBacks = managed.RemoteCallbacks()
```

Comment:
The above code is repeated everywhere. We can have a helper function to do the same, reducing the risk of making any mistakes in any of the checkouts.